weasyprint/layout/pages.py

```diff
@@ -511,6 +523,13 @@ def make_page(context, root_box, page_type, resume_at, content_empty):
     context.finish_block_formatting_context(root_box)
 
     page = page.copy_with_children([root_box])
+    descendants = list(page.descendants())
```

Comment:
This list might consume a lot of memory and doesn’t seem to serve any purpose. Please use the generator directly in the for-loop: `for child in page.descendants()`:.